docs/src/basemap.md

```diff
@@ -0,0 +1,81 @@
+```@meta
+CollapsedDocStrings = true
+```
+
+# Base maps
+
+A "base map" is essentially a static image composed of map tiles, accessible through the [`basemap`](@ref) function.  
+
+This function returns a tuple of `(x, y, image)`, where `x` and `y` are the dimensions of the image, and `image` is the image itself.  
+The image, and axes, are always in the Web Mercator coordinate system.
+
+While Tyler does not currently work with CairoMakie, this method is a way to add a basemap to a CairoMakie plot, though it will not update on zoom.
+
+```@docs; canonical=false
+basemap
+```
+
+## Examples
+
+
+Below are some cool examples of using basemaps.
+
+#### Cover example of London
+
+````@example coverlondon
+using Tyler, TileProviders
+
+xs, ys, img = basemap(
+    TileProviders.OpenStreetMap(),
+    Extent(X=(-0.5, 0.5), Y=(51.25, 51.75)),
+    (1024, 1024)
+)
+````
+
+````@example coverlondon
+using Makie
+image(xs, ys, img; axis = (; aspect = DataAspect()))
+````
+
+Note that the image is in the Web Mercator projection, as are the axes we see here.
+
+#### NASA GIBS tileset, plotted as a `meshimage` on a `GeoAxis`
+
+````@example nasagibs
+const BACKEND = Makie.current_backend() # hide
+using Tyler, TileProviders, GeoMakie, CairoMakie, Makie
+
+provider = TileProviders.NASAGIBS(:ViirsEarthAtNight2012)
+
+xs, ys, img = basemap(provider, Extent(X=(-90, 90), Y=(-90, 90)), (1024, 1024))
+````
+
+````@example nasagibs
+meshimage(
+    xs, ys, img; 
+    source = "+proj=webmerc", # REMEMBER: `img` is always in Web Mercator...
```

Comment:
would be nice if we kept crs and lookups attached - I think there is a package that does that ;)

Reply:
That's true :D - but then we'd have to add that as a dependency to Tyler as well.  

The other alternative is to add ImageIO and HTTP dependencies to MapTiles or TileProviders, and have them prompt individually.  But I don't think we want that...

Reply:
It would be nice to have a method of `basemap` that returned a raster at least, in an extension here or in Rasters.

Reply:
Question is how to trigger it. We could just pass the Raster type as the first argument?

Reply:
I would go the other way and create a constructor for `Raster(::TileProvider, extent; size, res, ...)`.

Reply:
This way it's at least a similar interface to what RasterDataSources has, for example

Reply:
Right, just call `basemap` internally to get the data. A bit like RasterDataSources.jl syntax

Reply:
Whats weird about that is it would need `using Tyler` which needs `Makie` when it really just needs ImageIO and HTTP

Reply:
It's the ImageIO part that gives me the shivers, since it's a lot of heavy binary dependenceis

Reply:
I'm more scared of the Makie precompile time for Plots.jl users